src/middlewared/middlewared/plugins/smb.py

```diff
@@ -339,6 +341,23 @@ def getparm(self, parm, section):
         except Exception as e:
             raise CallError(f'Attempt to query smb4.conf parameter [{parm}] failed with error: {e}')
 
+    @private
+    async def get_next_rid(self):
```

Comment:
Any chances for a race condition here? Do we need a lock?
I didnt follow the callers so I really dont know